src/explorer/CosmosDBTreeItem.ts

```diff
@@ -38,20 +38,38 @@ export class CosmosDBTreeItem extends AzureParentTreeItem<ISiteTreeRoot> {
                 label: 'Install Cosmos DB Extension...'
             })];
         }
-        const workspaceConfig = vscode.workspace.getConfiguration(constants.extensionPrefix);
-        const connections = workspaceConfig.get<IConnections[]>(constants.configurationSettings.connections, []);
-        // tslint:disable-next-line:strict-boolean-expressions
-        const unit = connections.find((x: IConnections) => x.webAppId === this.root.client.id) || <IConnections>{};
-        if (!unit.cosmosDB || unit.cosmosDB.length === 0) {
+
+        const connectionStrings: string[] = [];
+        const appSettingsNode = this.parent.parent.appSettingsNode;
+        const allAppSettings = await appSettingsNode.ensureSettings();
```

Comment:
No need to use the appSettingsNode directly. Just call `this.root.client.listApplicationSettings()`